### PR TITLE
[PATCH API-NEXT v2] Packet L3/L4 type implementation, alternative way

### PR DESCRIPTION
**Pull request by lumag**
This is a simple alternative to #459 implemented on @psavol 's request.

### CODE REVIEW

**Comment by lumag** [comment]
Repushed to trigger build

File: platform/linux-generic/odp_packet.c [hunk 1]
```diff
@@ -2544,3 +2544,41 @@ int odp_packet_has_ref(odp_packet_t pkt)
...
+		return ODP_PROTO_L3_TYPE_ARP;
+
+	return ODP_PROTO_L3_TYPE_NONE;
+}
+
+odp_proto_l4_type_t odp_packet_l4_type(odp_packet_t pkt)
+{
+	odp_packet_hdr_t *pkt_hdr = odp_packet_hdr(pkt);
+
+	if (pkt_hdr->p.input_flags.tcp)
```
Comment (bala-manoharan) [inline platform/linux-generic/odp_packet.c]: Why don't we use a switch case instead of a nested if-else?

Comment (muvarov) [reply platform/linux-generic/odp_packet.c]: @bala-manoharan swithc for set bits and mix of this bils? I don't think that it's possible here.

**Comment by muvarov** [comment]
commits are good with small rebase. There was discussion abot this PR. Waiting for resolution.

**Comment by lumag** [comment]
@muvarov which rebase? I don't see a conflict.

**Comment by psavol** [comment]
This is ready to be merged. Bala suggested to use switch-case but that does not work well with bit fields.

**Comment by lumag** [comment]
@muvarov can we please finally merge this, I have other pending patches.

**Comment by muvarov** [comment]
Merged.

**Pull request closed without merging**